### DISCUSSION TITLE
How to setup email in cloudlog?

### DISCUSSION THREAD

**Discussion by DJLax5** [magicbug/Cloudlog#2002] [Q&A]
Hi everyone!

I am new to cloudlog, and I am currently in the process of setting everything up. 
Sadly, I cannot get the email system to work. I have ssmtp installed on my system and I can send mails via the command line. But if I try to send a mail (e.g. by requesting a new password) from cloudlog, I get 
`Email settings are incorrect.`
Does this functionality depend on something else? The server runs within a docker container, so some packages may be missing. 
What should I set in crlf and newline?

I have already tried each combination of Outgoing Protocol and SMTP Encryption, the rest is correct. 

Any Ideas would be much appreciated!
Thanks!

**Comment by magicbug** [comment]
Settings are under /index.php/options/email or Admin -> Global -> Email out of the box its set for sendmail however that maybe not configured on your docker install, else you can provide it an SMTP server

**Reply by DJLax5** [reply]
Yeah, I know where the settings are. But they don't work.
The sendmail /ssmtp command on its own in the command line works fine.

**Comment by HB9HIL** [comment]
Same here
My E-Mail Server Settings using SMTP are correct. They work in other applications absolutely without problems. 

But Cloudlog tells me the E-Mail Settings are incorrect with following Error Log:

`Severity: Notice --> fwrite(): send of 6 bytes failed with errno=32 Broken pipe /var/www/html/system/libraries/Email.php 2217`

Could someone help me? I don't want to use sendmail.

**Comment by magicbug** [comment]
This error is usually related to SMTP Encryption

**Reply by HB9HIL** [reply]
I use SSL on Port 465. Other Applications work without Problems..

**Comment by HB9HIL** [comment]
Still not getting it to work. The mail settings are correct but cloudlog always tells me E-Mail Settings are not correct.

**Comment by HB9HIL** [comment]
> Hi everyone!
> 
> I am new to cloudlog, and I am currently in the process of setting everything up. Sadly, I cannot get the email system to work. I have ssmtp installed on my system and I can send mails via the command line. But if I try to send a mail (e.g. by requesting a new password) from cloudlog, I get `Email settings are incorrect.` Does this functionality depend on something else? The server runs within a docker container, so some packages may be missing. What should I set in crlf and newline?
> 
> I have already tried each combination of Outgoing Protocol and SMTP Encryption, the rest is correct.
> 
> Any Ideas would be much appreciated! Thanks!

Did you make some progress with that?

**Reply by DJLax5** [reply]
Sorry, No. 
But I also did not make any in depth resarch as to why, as i don't need the mail functionality. I just wanted to configure it, as CL offers it.

**Comment by HB9HIL** [comment]
My Mailserver says this:

`warning: TLS library problem: error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../ssl/record/ssl3_record.c:331:`

**Comment by HB9HIL** [comment]
After some research (I'm no developer)...

Could be the Problem the old CodeIgniter 3 Framework?... CodeIgniter 4 works much better with the newer PHP Versions.. 

For me it seems like that system/libraries/Email.php is the Problem which causes the issues with the SMTP Mailer. Sendmail is also nice but SMTP is a bit easier to configure and if Cloudlog offers SMTP directly so why not using it..

**Reply by AndreasK79** [reply]
It is difficult to say where the problem is, but it works with the correct settings. I got it to work locally with my gmail, but not on my host, so something is different with the servers.

Porting Cloudlog over to CodeIgniter 4 will probably never happen, as it is a lot of work. At least we are trying to keep Cloudlog (with CI 3) working with newer PHP versions.

**Comment by HB9HIL** [comment]
@DJLax5 Finally I found the problem. The PR #2715 should fix this.

**Reply by DJLax5** [reply]
Nice! Will take a look! Thanks!

**Comment by MikleBas** [comment]
Hi all!
I want to ask a question - what exactly and where will the cloudlog be sent? Why does he need the mail server settings?

**Reply by GregTroar** [reply]
Same question here what is the email server used for ?
Which emails are sent by Cloudlog ??

**Reply by HB9HIL** [reply]
At the moment Cloudlog is using the Mail Settings for 2 Types of Mails:


If E-Mail is set properly

- Passwort Reset (Forgot Password on Login Page)
- OQRS Notifications if a new OQRS Request was placed on the Public Log

**Comment by GregTroar** [comment]
ok great thanks for info. Won't use any of these options so no need for me.

Thanks
